test/contracts/kill-switch/kill_switch.js

```diff
@@ -0,0 +1,972 @@
+                })
+              })
+            })
+
+            context('when the bug was a false positive', () => {
+              beforeEach('roll back reported bug', async () => {
+                await defaultIssuesRegistry.setSeverityFor(appBase.address, SEVERITY.NONE, { from: securityPartner })
+              })
+
+              context('when there is no highest allowed severity set for the contract being called', () => {
+                itExecutesTheCallUnlessInstanceNotWhitelistedAndBaseBlacklisted()
+              })
+
+              context('when there is a highest allowed severity set for the contract being called', () => {
+                context('when the highest allowed severity is under the reported bug severity', () => {
+                  beforeEach('set highest allowed severity below the one reported', async () => {
+                    await killSwitch.setHighestAllowedSeverity(SAMPLE_APP_ID, SEVERITY.LOW, { from: owner })
+                  })
+
+                  itExecutesTheCallUnlessInstanceNotWhitelistedAndBaseBlacklisted()
+                })
+
+                context('when the highest allowed severity is equal to the reported bug severity', () => {
+                  beforeEach('set highest allowed severity equal to the one reported', async () => {
+                    await killSwitch.setHighestAllowedSeverity(SAMPLE_APP_ID, SEVERITY.MID, { from: owner })
+                  })
+
+                  itExecutesTheCallUnlessInstanceNotWhitelistedAndBaseBlacklisted()
+                })
+
+                context('when the highest allowed severity is greater than the reported bug severity', () => {
+                  beforeEach('set highest allowed severity above the one reported', async () => {
+                    await killSwitch.setHighestAllowedSeverity(SAMPLE_APP_ID, SEVERITY.CRITICAL, { from: owner })
+                  })
+
+                  itExecutesTheCallUnlessInstanceNotWhitelistedAndBaseBlacklisted()
+                })
+              })
+            })
+          })
+        })
+      })
+
+      describe('gas costs', () => {
+        beforeEach('set a highest allowed severity', async () => {
+          await killSwitch.setHighestAllowedSeverity(SAMPLE_APP_ID, SEVERITY.MID, { from: owner })
+          await defaultIssuesRegistry.setSeverityFor(appBase.address, SEVERITY.LOW, { from: securityPartner })
+        })
+
+        it('kill switch should overload ~27k of gas to a function', skipCoverage(async () => {
+          const { receipt: { cumulativeGasUsed: gasUsedWithKillSwitch } } = await app.write(10, { from: owner })
+          const { receipt: { cumulativeGasUsed: gasUsedWithoutKillSwitch } } = await app.writeWithoutKillSwitch(10, { from: owner })
+
+          const killSwitchCost = gasUsedWithKillSwitch - gasUsedWithoutKillSwitch
+          assert(killSwitchCost <= 27000, 'kill switch should overload ~27k of gas')
```

Comment:
Nitpick: wondering if there was a reason to not use [`assert.isAtMost()`](https://www.chaijs.com/api/assert/#method_isatmost) here, so the tests would log the actual gas used on failure?